src/ts-serialize.ts

```diff
@@ -34,15 +35,42 @@ export function serializeTsDeclarations(
       return serializeProperty(node, depth);
     default:
       const never: never = node;
-      throw new Error(`Unknown node kind: ${util.inspect(never)}`);
+      throw new Error(`Unknown declarations AST kind: ${util.inspect(never)}`);
   }
 }
 
-function serializeDocument(node: Document): string {
+/**
+ * Serialize a TypeScript type expression AST to a string.
+ */
+export function serializeType(node: ts.Type, greedy = false): string {
+  switch (node.kind) {
+    case 'name':
+      return node.name;
+
+    case 'array':
+      return `Array<${serializeType(node.itemType)}>`;
```

Comment:
With this additional abstraction you could now easily produce idiomatic code by checking whether node.itemType.kind is `name`, in which case you could do `${serializeType(node.itemType)}[]`

Reply:
Done.